cmd/flow-dps-live/main.go

```diff
@@ -0,0 +1,411 @@
+			Str("peer_address", flagPeerAddr).
+			Msg("invalid peer address format")
+		return failure
+	}
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("peer_address", flagPeerAddr).
+			Msg("invalid peer port")
+		return failure
+	}
+	key, err := crypto.DecodePublicKeyHex(crypto.ECDSA_P256, flagPeerKey)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("peer_key", flagPeerKey).
+			Msg("invalid peer public key")
+		return failure
+	}
+	// FIXME: Get public keys from transit stuff.
+	bootstrapIdentities := []follower.BootstrapNodeInfo{{
+		Host:             host,
+		Port:             uint(port),
+		NetworkPublicKey: key,
+	}}
+	follower, err := follower.NewConsensusFollower(nodeID, bootstrapIdentities, flagBindAddr, follower.WithDataDir(flagData))
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("bucket", flagBucket).
+			Msg("could not create consensus follower")
+		return failure
+	}
+
+	execution := execution.New(log, downloader, codec, data)
+	chain := chain.FromFollower(log, execution, data)
+
+	// FIXME: Is the consensus needed anywhere besides existing to make it index things
+	//  and call the exec follower's callback.
+	_ = consensus.New(log, execution, follower, data)
+
+	// Writer is responsible for writing the index data to the index database.
+	writer := index.NewWriter(db, storage)
+	defer func() {
+		err := writer.Close()
+		if err != nil {
+			log.Error().Err(err).Msg("could not close index")
+		}
+	}()
+	write := dps.Writer(writer)
+	if flagMetrics {
+		time := rcrowley.NewTime("write")
+		mout.Register(time)
+		write = metrics.NewWriter(write, time)
+	}
+
+	// Initialize the transitions with the dependencies and add them to the FSM.
+	transitions := mapper.NewTransitions(log, load, chain, execution, write,
+		mapper.WithIndexCommit(flagIndexAll || flagIndexCommit),
+		mapper.WithIndexHeader(flagIndexAll || flagIndexHeader),
+		mapper.WithIndexCollections(flagIndexAll || flagIndexCollections),
+		mapper.WithIndexGuarantees(flagIndexAll || flagIndexGuarantees),
+		mapper.WithIndexTransactions(flagIndexAll || flagIndexTransactions),
+		mapper.WithIndexResults(flagIndexAll || flagIndexResults),
+		mapper.WithIndexEvents(flagIndexAll || flagIndexEvents),
+		mapper.WithIndexPayloads(flagIndexAll || flagIndexPayloads),
+		mapper.WithIndexSeals(flagIndexAll || flagIndexSeals),
+		mapper.WithSkipBootstrap(flagSkipBootstrap),
+	)
+	forest := forest.New()
+	state := mapper.EmptyState(forest)
+	fsm := mapper.NewFSM(state,
+		mapper.WithTransition(mapper.StatusEmpty, transitions.BootstrapState),
+		mapper.WithTransition(mapper.StatusUpdating, transitions.UpdateTree),
+		mapper.WithTransition(mapper.StatusMatched, transitions.CollectRegisters),
+		mapper.WithTransition(mapper.StatusCollected, transitions.IndexRegisters),
+		mapper.WithTransition(mapper.StatusIndexed, transitions.ForwardHeight),
+		mapper.WithTransition(mapper.StatusForwarded, transitions.IndexChain),
+	)
```

Comment:
I guess we could have this part different between the file and live indexer, in order to add the wait state here. We would just need to losen up some of the sanity checks on transitions inside the transitions file, but they were a defensive redundant check anyway.

Reply:
You mean adding a transition to a `mapper.StatusWaiting`, right?